datetimepicker.js

```diff
@@ -132,12 +147,32 @@ angular.module('ui.bootstrap.datetimepicker',
           }
         ],
         link: function(scope, element) {
+          var firstTimeAssign = true;
+
           scope.$watch(function() {
             return scope.ngModel;
-          }, function(ngModel) {
+          }, function(newTime) {
             // if a time element is focused, updating its model will cause hours/minutes to be formatted by padding with leading zeros
             if (!element.children()[1].contains(document.activeElement)) {
-              scope.time = new Date(ngModel);
+
+              if (newTime == null || newTime === '') { // if the newTime is not defined
+                if (firstTimeAssign) { // if it's the first time we assign the time value
+                  // create a new default time where the hours, minutes, seconds and milliseconds are set to 0.
+                  newTime = new Date;
+                  newTime.setHours(0, 0, 0, 0);
```

Comment:
Previously, with `new Date(ngModel)` when `ngModel == null`, you'd get a date object created from Epoch time at midnight UTC. When a user's computer is configured with a timezone of, say +11 hours like here in Sydney, you'd get a default time of 11:00. 

The thing to bear in mind is that the date is empty/null atm. So, to make it clear that the time is somewhat at a "default" state, I chose to set it to `00:00`. 

Should we perhaps propose a new option to set the default time? 
e.g. 
- current time 
- given default time 

As a side note, that's also why I've raised another PR to propose to hide the time inputs when the date is null/empty.